packages/storage/compat/reference.ts

```diff
@@ -114,8 +115,17 @@ export class ReferenceCompat implements types.Reference {
     metadata?: Metadata
   ): types.UploadTask {
     this._throwIfRoot('putString');
+    const data = dataFromString(format, value);
+    const metadataClone = { ...metadata } as Metadata;
+    if (metadataClone['contentType'] == null && data.contentType != null) {
+      metadataClone['contentType'] = data.contentType!;
```

Comment:
Is the bang needed here? Doesn't the `!=` check catch both `null` and `undefined`?

Reply:
Removed, seems like it was left over from back when it was using `type.isDef` instead of `!= null`.